docs/source/Developer/MVP_guidelines.rst

```diff
@@ -0,0 +1,207 @@
+
+.. _MVP_Guidelines:
+
+=====================================
+Model-View-Presenter (MVP) Guidelines
+=====================================
+
+Overview
+========
+
+The Model-View-Presenter (MVP) pattern is a software design pattern
+that separates the structure of an application into three main components:
+Model, View, and Presenter. This separation enhances modularity, testability,
+and maintainability of the codebase.
+
+Components
+===========
+
+1. **Model:**
+   The Model represents the application's data and algorithm logic. It is
+   responsible for retrieving and manipulating data, as well as notifying the
+   Presenter of any changes. The Model is independent of both the View and the
+   Presenter.
+
+2. **View:**
+   The View is responsible for presenting the data to the user and receiving
+   user input. It remains passive and delegates user interactions to the Presenter.
+   The View does not contain any application logic and is only concerned with
+   displaying information to the user. Each View is associated with a single Presenter.
+
+3. **Presenter:**
+   The Presenter acts as an intermediary between the Model and the View. It
+   contains the application logic, responds to user input from the View, and
+   updates the Model accordingly. The Presenter also updates the View in response
+   to changes in the Model. This separation allows for easier testing and
+   maintenance of the application logic. There is only one Presenter for each View.
+
+Communication Flow
+===================
+
+1. User interacts with the View by triggering events.
+2. The View forwards user input events to the Presenter.
+3. The Presenter processes the events, updates the Model as needed, and
+   communicates changes back to the View.
+4. The View reflects the changes made by the Presenter.
+
+.. mermaid::
+
+    graph LR
+    User((User))
+    View(View)
+    Presenter(Presenter)
+    Model(Model)
+
+    User -->|User Interaction| View
+    View -->|User Input Events| Presenter
+    Presenter -->|Update Model| Model
+    Model -->|Notify Changes| Presenter
+    Presenter -->|Update View| View
+    View -->|View Updates| User
+
+
+|
+
+Guidelines
+===========
+
+Model
+`````
+
+1. The Model should be independent of the View and the Presenter.
+2. The Model should contain the application's data and algorithm logic.
+3. The Model should be reusable and not tied to a specific View or Presenter.
+4. Absolutely no UI logic should be present in the Model.
+
+View
+````
+1. The View should be passive and not contain any application logic.
+2. The View should delegate user interactions to the Presenter.
+3. The View should be responsible for displaying data to the user and receiving user input.
+4. The View should not contain any logic or data manipulation.
```

Comment:
I would replace this with event triggering and simple field validations.
there should be some logic and data handling.